airflow/providers/google/cloud/hooks/compute.py

```diff
@@ -72,6 +76,350 @@ def get_conn(self):
+            and network_interfaces and etc will be passed to Instance as they are specified
+            in the Instance Template.
+            Full or partial URL and can be represented as examples below:
+            1. https://www.googleapis.com/compute/v1/projects/your-project-name/global/instanceTemplates/
+               your-instanceTemplate-name
+            2. projects/your-project-name/global/instanceTemplates/your-instanceTemplate-name
+            3. global/instanceTemplates/your-instanceTemplate-name
+        :param request_id: Unique request_id that you might add to achieve
+            full idempotence (for example when client call times out repeating the request
+            with the same request id will not create a new instance template again)
+            It should be in UUID format as defined in RFC 4122
+        :param retry: A retry object used  to retry requests. If `None` is specified, requests
+            will not be retried.
+        :param timeout: The amount of time, in seconds, to wait for the request to complete.
+            Note that if `retry` is specified, the timeout applies to each individual attempt.
+        :param metadata: Additional metadata that is provided to the method.
+        """
+        client = self.get_compute_instance_client()
+        client.insert(
+            # Calling method insert() on client to create Instance.
+            # This method accepts request object as an argument and should be of type
+            # Union[google.cloud.compute_v1.types.InsertInstanceRequest, dict] to construct a request
+            # message.
+            # The request object should be represented using arguments:
+            #   instance_resource (google.cloud.compute_v1.types.Instance):
+            #       The body resource for this request.
+            #   request_id (str):
+            #       Optional, request ID to identify requests.
+            #   project (str):
+            #       Project ID for this request.
+            #   zone (str):
+            #       The name of the zone for this request.
+            #   source_instance_template (str):
+            #       Optional, link to Instance Template, that can be used to create new Instance.
+            request={
+                'instance_resource': body,
+                'request_id': request_id,
+                'project': project_id,
+                'zone': zone,
+                'source_instance_template': source_instance_template,
+            },
+            retry=retry,
+            timeout=timeout,
+            metadata=metadata,
+        )
+
+    @GoogleBaseHook.fallback_to_default_project_id
+    def get_instance(
+        self,
+        resource_id: str,
+        zone: str,
+        project_id: str = PROVIDE_PROJECT_ID,
+        retry: Optional[Retry] = None,
+        timeout: Optional[float] = None,
+        metadata: Optional[Sequence[Tuple[str, str]]] = (),
+    ) -> dict:
```

Comment:
The same story, I think, return type annotation should be Instance Object not a dict

Reply:
Done